Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -75,6 +75,26 @@ struct EFixedProvider {
 
 namespace Mantid::Algorithms {
 
+std::unique_ptr<DiscusData2D> DiscusData2D::CreateCopy(bool clearY) {
+  auto data2DNew = std::make_unique<DiscusData2D>();
+  data2DNew->m_data.resize(m_data.size());
+  for (size_t i = 0; i < m_data.size(); i++) {
+    data2DNew->m_data[i].X = m_data[i].X;
+    if (clearY)
+      data2DNew->m_data[i].Y = std::vector<double>(m_data[i].Y.size(), 0.);
+    else
+      data2DNew->m_data[i].Y = m_data[i].Y;
```

Comment:
```suggestion
    data2DNew->m_data[i].Y = clearY ? std::vector<double>(m_data[i].Y.size(), 0.) : m_data[i].Y;
```
Just a suggestion, no need to change if you think its not as easy to read